code/modules/recycling/disposal/holder.dm

```diff
@@ -41,19 +41,22 @@
 	// now everything inside the disposal gets put into the holder
 	// note AM since can contain mobs or objs
 	for(var/A in D)
-		var/atom/movable/AM = A
-		if(AM == src)
+		var/atom/movable/atom_movable = A
+		if(atom_movable == src)
 			continue
-		SEND_SIGNAL(AM, COMSIG_MOVABLE_DISPOSING, src, D)
-		AM.forceMove(src)
-		if(istype(AM, /obj/structure/big_delivery) && !hasmob)
-			var/obj/structure/big_delivery/T = AM
-			src.destinationTag = T.sortTag
-		else if(istype(AM, /obj/item/small_delivery) && !hasmob)
-			var/obj/item/small_delivery/T = AM
-			src.destinationTag = T.sortTag
-		else if(istype(AM, /mob/living/silicon/robot))
-			var/obj/item/dest_tagger/borg/tagger = locate() in AM
+		SEND_SIGNAL(atom_movable, COMSIG_MOVABLE_DISPOSING, src, D)
+		atom_movable.forceMove(src)
+		if(istype(atom_movable, /obj/structure/big_delivery) && !hasmob)
```

Comment:
I think we're at the point where this should be modularized into just having all these register signals.